src/plugins/data_source/server/legacy/http_aws_es/connector.js

```diff
@@ -91,15 +92,25 @@ class HttpAmazonESConnector extends HttpConnector {
   }
 
   createRequest(params, reqParams) {
+    const [pathname = '/', queryStr = ''] = (reqParams.path || '').split('?', 2);
+
+    const queryParams = queryStr ? queryString.parse(queryStr) : undefined;
+
     const request = new HttpRequest({
       ...this.endpoint,
       method: reqParams.method,
       headers: reqParams.headers || {},
       hostname: this.endpoint.hostname,
+      query: queryParams,
     });
 
-    // copy across params
-    Object.assign(request, reqParams);
+    Object.assign(request, {
+      ...reqParams,
+      path: pathname.replaceAll(
+        /[!'()*]/g,
+        (c) => `%${c.charCodeAt(0).toString(16).toUpperCase()}`
+      ),
+    });
```

Comment:
Nit: a comment telling us:
- `new HttpRequest` will erase the query field.
-  Follow `https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#encoding_for_rfc3986` to encode special chars

here would be much helpful